pkg/controller/core/core.go

```diff
@@ -29,25 +29,12 @@ import (
 )
 
 const (
-	updateChBuffer                   = 10
-	defaultRequeuingBaseDelaySeconds = 10
+	updateChBuffer = 10
 )
 
-type ControllerOptions struct {
-	requeuingBaseDelaySeconds int32
-}
-
-type ControllerOption func(*ControllerOptions)
-
-func WithControllerRequeuingBaseDelaySeconds(value int32) ControllerOption {
-	return func(o *ControllerOptions) {
-		o.requeuingBaseDelaySeconds = value
-	}
-}
-
 // SetupControllers sets up the core controllers. It returns the name of the
 // controller that failed to create and an error, if any.
-func SetupControllers(mgr ctrl.Manager, qManager *queue.Manager, cc *cache.Cache, cfg *configapi.Configuration, controllerOpts ...ControllerOption) (string, error) {
+func SetupControllers(mgr ctrl.Manager, qManager *queue.Manager, cc *cache.Cache, cfg *configapi.Configuration) (string, error) {
```

Comment:
I personally like the `Options` pattern better.

Reply:
Right, but the new parameter is not needed any longer since the configuration is not passed in ` cfg *configapi.Configuration`. I guess we could replace  ` cfg *configapi.Configuration` with `Options`, but it would be a lot of changes, so I wouldn't like to mix this into this PR.

Reply:
for now changing `WithControllerRequeuingBaseDelaySeconds` to something like `WithControllerRequeuingBackoffBaseSeconds`  can work and we can do a clean-up to drop the cfg usage.

Reply:
I see that, but the interim state of passing the Paramus both ways would look weird. I'm happy to open a dedicated issue to do the refactoring at once.

Reply:
If we agree we should go towards `Options` , why start with a step back?

Reply:
 the interim state of passing the params which are part of the cfg both ways would look weird

Reply:
And also I'm not 100% percent sure it will be better. Someone had a reason to make it like that in the first place, so I think it will require an actual experiment to see if there are no unforeseen complications

Reply:
It looks like dropping cfg would require changing a lot of functions (all the SetupWithManager) calls.
It might be worth exploring, but I agree that it's too much for this PR.

Reply:
In this PR there is the option to pass the new param as option just like the old one was set.

Reply:
Yes, but the cfg is already included.

If options is the path forward, it's beyond the scope of this PR.

Reply:
We are dropping the ControllerOption type in this PR, we should not do that.

Reply:
Right now, there is no use for it. If we can make a case for bringing it back, we will.

Reply:
> going the path of keeping it requires more changes

I'm not convinced, we used to have for now changing `WithControllerRequeuingBaseDelaySeconds`  we can just replace it with  `WithControllerRequeuingBackoffBaseSeconds` .

Reply:
But then we have the same information flowing through `cfg`

Reply:
Let me put the context of the reason why we added the `controllerOpts`.

We introduced the `controllerOpts` in https://github.com/kubernetes-sigs/kueue/pull/2025 instead of Config API to cherry-pick to the release v0.6 as an interim approach. Then, we try to remove the interim approach in this PR.

I agree with refactoring this function, but as @alculquicondor and @mimowo, I also think that the refactoring is beyond this PR responsibility.
So, I agree with leaving the refactoring in another PR.